### PR TITLE
feat: add muted hyperlinks to list view

### PR DESCRIPTION
**Pull request by katrinan029**
# Description
As part of enhancing the user experience in the provisioning tool for LC2, we need to implement a List View of Learner Credit Plans that includes hyperlinks. This List View should allow users to access the detailed view of any selected plan through muted standalone hyperlinks on the Plan ID, Plan Name, and Customer Name fields.

# Solution
Add Paragon's `Hyperlink` component to Plan ID, Plan Name, and Customer Name fields in the List View. The link navigates the user to a detailed plan view that will be implemented in the subsequent PR.

https://github.com/openedx/frontend-app-support-tools/assets/71999631/bc6823eb-354f-4e74-8486-88c0eb3a1a1a

Enable feature flag `FEATURE_CONFIGURATION_EDIT_ENTERPRISE_PROVISION` to see these changes:
https://support-tools.stage.edx.org/enterprise-configuration/learner-credit?feature=FEATURE_CONFIGURATION_EDIT_ENTERPRISE_PROVISION

[JIRA TICKET](https://2u-internal.atlassian.net/browse/ENT-7501)
[Figma mocks](https://www.figma.com/file/JpAMqHcvr5OpzYPFsQopZG/LC2%3A-Enterprise-Setup-tool-Modifications?node-id=266%3A12265&mode=dev)

### CODE REVIEW

File: src/Configuration/Provisioning/DashboardDataTable/tests/DashboardTableLinks.test.jsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    });
+
+    it('does not render hyperlink if FEATURE_CONFIGURATION_EDIT_ENTERPRISE_PROVISION is false', () => {
+      getConfig.mockReturnValue({ FEATURE_CONFIGURATION_EDIT_ENTERPRISE_PROVISION: false });
+      render(<PlanIdHyperlink row={row} />);
+      expect(screen.queryByRole('link', { name: '123456789' })).toBeNull();
+    });
+  });
+
+  describe('PlanIdHyperlink', () => {
```
Comment (marlonkeating) [inline src/Configuration/Provisioning/DashboardDataTable/tests/DashboardTableLinks.test.jsx]:
```suggestion
  describe('PlanTitleHyperlink ', () => {
```

File: src/Configuration/Provisioning/DashboardDataTable/DashboardTableLinks.jsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        destination={`${HOME}/${planRowUuid}/view`}
+        key={`edit-${planRowUuid}`}
+        variant="muted"
+      >
+        {planRowUuid}
+      </Hyperlink>
+    );
+  }
+
+  return planRowUuid;
```
Comment (marlonkeating) [inline src/Configuration/Provisioning/DashboardDataTable/DashboardTableLinks.jsx]: nit: Given the hyperlink for 3 out of the 4 rows is largely the same except for the text, we could shorten these up with a helper method.
```suggestion
  dashboardLink = (planRowUuid, title) => {
    if (getConfig().FEATURE_CONFIGURATION_EDIT_ENTERPRISE_PROVISION) {
      return (
        <Hyperlink
          destination={`${HOME}/${planRowUuid}/view`}
          key={`edit-${planRowUuid}`}
          variant="muted"
        >
          {title}
        </Hyperlink>
      );
    }
    return title;
  }

  return dashboardLink(row.values.uuid, row.values.uuid);
  }
  
```

**Review by marlonkeating — CHANGES_REQUESTED** [review]

File: src/Configuration/Provisioning/DashboardDataTable/tests/DashboardTableLinks.test.jsx (hunk 1 above)
Comment (katrinan029) [reply src/Configuration/Provisioning/DashboardDataTable/tests/DashboardTableLinks.test.jsx]: good catch!

File: src/Configuration/Provisioning/DashboardDataTable/DashboardTableLinks.jsx (hunk 2 above)
Comment (katrinan029) [reply src/Configuration/Provisioning/DashboardDataTable/DashboardTableLinks.jsx]: This was a great suggestion. I updated my PR to incorporate this. Thank you!

**Review by marlonkeating — APPROVED** [review]

**Pull request merged**